.buildkite/pipeline.yml

```
@@ -1,4 +1,69 @@
+env:
+  DOCKER_HUB_REPO: 'forumone/gesso'
+
+# Consolidate the common configuration for reuse across multiple steps.
+definitions:
+  # Configure some default step values.
+  step-defaults: &step-defaults
+    timeout_in_minutes: 15
+  matrix-setup: &matrix-setup
+    setup:
+      php:
+        - "7.4"
+        - "8.0"
+      node_version:
+        - "16"
+        - "14"
+        - "12"
+      gesso_version:
+        - "4"
+        - "3"
+  docker-build: &docker-build
+    - "docker build -t $DOCKER_HUB_REPO:{{matrix.gesso_version}}-node-v{{matrix.node_version}}-php-{{matrix.php}} --build-arg PHP_VERSION={{matrix.php}} --build-arg NODE_VERSION={{matrix.node_version}} ."
+  docker-push: &docker-push
+    - "docker push $DOCKER_HUB_REPO:{{matrix.gesso_version}}-node-v{{matrix.node_version}}-php-{{matrix.php}}"
+  queues:
+    # Docker-based tasks should run in the Docker queue.
+    docker-agents: &docker-agents
+      agents:
+        queue: "docker-builders"
+  plugins:
+    seek: &seek
+      seek-oss/aws-sm#v2.0.0:
+        env:
+          DOCKER_LOGIN_PASSWORD: buildkite/dockerhubid
+    docker-login: &docker-login
+      docker-login#v2.0.1:
+        username: f1builder
+        password-env: DOCKER_LOGIN_PASSWORD
+
 steps:
-  - label: ":pipeline:"
-    commands:
-      - bash .buildkite/pipeline.sh | buildkite-agent pipeline upload
+  - label: ":pipeline: Build :php: {{matrix.gesso_version}}-node-v{{matrix.node_version}}-php-{{matrix.php}}"
+    <<: *docker-agents
```

Let's add concurrency limits so we don't hog the agent pool.

```suggestion
    <<: *docker-agents
    concurrency_group: $BUILDKITE_PIPELINE_SLUG/docker
    concurrency: 4
```